scripts/common/rook.sh

```diff
@@ -173,3 +173,238 @@ function maybe_cleanup_rook() {
+    tools_pod="$(kubectl -n rook-ceph get pod -l "app=rook-ceph-tools" -o jsonpath='{.items[0].metadata.name}')"
+    if kubectl -n rook-ceph exec "${tools_pod}" -- bash -s "$args" ; then
+        return 0
+    fi
+    return 1
+}
+
+# upgrades Rook progressively from 1.0.x to 1.4.x
+function rook_10_to_14() {
+    logStep "Upgrading Rook-Ceph from 1.0.x to 1.4.x"
+    echo "This involves upgrading from 1.0.x to 1.1, 1.1 to 1.2, 1.2 to 1.3, and 1.3 to 1.4"
+    echo "This may take some time"
+
+    $DIR/bin/kurl rook hostpath-to-block
+
+    logStep "Downloading images required for this upgrade"
+    # todo download images and load them so that they aren't being pulled dynamically
+    logSuccess "Images loaded for Rook 1.1.9, 1.2.7, 1.3.11 and 1.4.9"
+
+    echo "Rescaling pgs per pool"
+    rook_ceph_tools_exec "ceph osd pool ls | grep rook-ceph-store | xargs -I {} ceph osd pool set {} pg_num 16"
+    rook_ceph_tools_exec "ceph osd pool ls | grep -v rook-ceph-store | xargs -I {} ceph osd pool set {} pg_num 32"
+    $DIR/bin/kurl rook wait-for-health
+
+    logStep "Upgrading to Rook 1.1.9"
+    $DIR/bin/kurl rook wait-for-health
+
+    # first update rbac and other resources for 1.1
+    # todo store these files ourselves
+    curl -sSL https://raw.githubusercontent.com/rook/rook/v1.1.9/cluster/examples/kubernetes/ceph/upgrade-from-v1.0-create.yaml \
+      | sed 's/ROOK_SYSTEM_NAMESPACE/rook-ceph/g' | sed 's/ROOK_NAMESPACE/rook-ceph/g' | kubectl create -f -
+    curl -sSL https://raw.githubusercontent.com/rook/rook/v1.1.9/cluster/examples/kubernetes/ceph/upgrade-from-v1.0-apply.yaml \
+      | sed 's/ROOK_SYSTEM_NAMESPACE/rook-ceph/g' | sed 's/ROOK_NAMESPACE/rook-ceph/g' | kubectl apply -f -
+
+    kubectl delete crd volumesnapshotclasses.snapshot.storage.k8s.io volumesnapshotcontents.snapshot.storage.k8s.io volumesnapshots.snapshot.storage.k8s.io
+    kubectl -n rook-ceph set image deploy/rook-ceph-operator rook-ceph-operator=rook/ceph:v1.1.9
+    kubectl -n rook-ceph set image deploy/rook-ceph-tools rook-ceph-tools=rook/ceph:v1.1.9
+    echo "Waiting for Rook 1.1.9 to rollout throughout the cluster, this may take some time"
+    # todo show progress somehow
+    spinner_until -1 is_rook_rollout_complete "v1.1.9"
+    # todo make sure that the RGW isn't getting stuck
+    echo "Rook 1.1.9 has been rolled out throughout the cluster"
+
+    echo "Enabling pg pool autoscaling"
+    rook_ceph_tools_exec "ceph osd pool ls | xargs -I {} ceph osd pool set {} pg_autoscale_mode on"
+
+    echo "Upgrading CRDs to Rook 1.1"
+    # todo store these files ourselves
+    curl -sSL https://raw.githubusercontent.com/rook/rook/v1.1.9/cluster/examples/kubernetes/ceph/upgrade-from-v1.0-crds.yaml | kubectl apply -f -
+
+    # todo upgrade ceph image to 14.2.5 here
+    echo "Upgrading ceph to v14.2.5"
+    kubectl -n rook-ceph patch CephCluster rook-ceph --type=merge -p "{\"spec\": {\"cephVersion\": {\"image\": \"ceph/ceph:v14.2.5-20191210\"}}}"
```

Comment:
```suggestion
    kubectl -n rook-ceph patch CephCluster rook-ceph --type=merge -p '{"spec": {"cephVersion": {"image": "ceph/ceph:v14.2.5-20191210"}}}'
```

Reply:
I think I copied that from the ceph docs directly
single quotes are much better